PostHog/Models/Survey.swift

```diff
@@ -0,0 +1,396 @@
+//
+//  Survey.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 20/01/2025.
+//
+
+import Foundation
+
+/// Represents the main survey object containing metadata, questions, conditions, and appearance settings.
+/// see: posthog-js/posthog-surveys-types.ts
+struct Survey: Decodable, Identifiable {
+    /// The unique identifier for the survey
```

Comment:
2 things here.
I think we should break into different files for most of the models.
Also, consider adding a PostHog prefix because of name clashing.